xbmc/utils/SystemInfo.cpp

```diff
@@ -451,67 +453,64 @@ const std::string& CSysInfo::GetAppName(void)
 
 bool CSysInfo::GetDiskSpace(const std::string& drive,int& iTotal, int& iTotalFree, int& iTotalUsed, int& iPercentFree, int& iPercentUsed)
 {
-  bool bRet= false;
-  ULARGE_INTEGER ULTotal= { { 0 } };
-  ULARGE_INTEGER ULTotalFree= { { 0 } };
+  using namespace KODI::PLATFORM::FILESYSTEM;
+
+  bool bRet = false;
+  space_info total = { 0 };
+  std::error_code ec;
 
-  if( !drive.empty() && drive != "*" )
+  if (!drive.empty() && drive != "*")
   {
 #ifdef TARGET_WINDOWS
-    UINT uidriveType = GetDriveType(( drive + ":\\" ).c_str());
-    if(uidriveType != DRIVE_UNKNOWN && uidriveType != DRIVE_NO_ROOT_DIR)
-      bRet= ( 0 != GetDiskFreeSpaceEx( ( drive + ":\\" ).c_str(), NULL, &ULTotal, &ULTotalFree) );
+    UINT uidriveType = GetDriveType((drive + ":\\").c_str());
+    if (uidriveType != DRIVE_UNKNOWN && uidriveType != DRIVE_NO_ROOT_DIR)
+      total = space(drive + ":\\", ec);
 #elif defined(TARGET_POSIX)
-    bRet = (0 != GetDiskFreeSpaceEx(drive.c_str(), NULL, &ULTotal, &ULTotalFree));
+    total = space(drive, ec);
 #endif
+    bRet = ec.value() == 0;
   }
   else
   {
-    ULARGE_INTEGER ULTotalTmp= { { 0 } };
-    ULARGE_INTEGER ULTotalFreeTmp= { { 0 } };
 #ifdef TARGET_WINDOWS
-    char* pcBuffer= NULL;
-    DWORD dwStrLength= GetLogicalDriveStrings( 0, pcBuffer );
-    if( dwStrLength != 0 )
+    space_info si;
+    char* pcBuffer = nullptr;
+    DWORD dwStrLength = GetLogicalDriveStrings(0, pcBuffer);
+    if (dwStrLength != 0)
     {
-      dwStrLength+= 1;
-      pcBuffer= new char [dwStrLength];
-      GetLogicalDriveStrings( dwStrLength, pcBuffer );
-      int iPos= 0;
+      dwStrLength += 1;
+      pcBuffer = new char[dwStrLength];
+      GetLogicalDriveStrings(dwStrLength, pcBuffer);
+      int iPos = 0;
       do {
-        if( DRIVE_FIXED == GetDriveType( pcBuffer + iPos  ) &&
-            GetDiskFreeSpaceEx( ( pcBuffer + iPos ), NULL, &ULTotal, &ULTotalFree ) )
+        if (DRIVE_FIXED == GetDriveType(pcBuffer + iPos))
         {
-          ULTotalTmp.QuadPart+= ULTotal.QuadPart;
-          ULTotalFreeTmp.QuadPart+= ULTotalFree.QuadPart;
+          si = space(pcBuffer + iPos, ec);
+          if (ec.value() == 0)
+          {
+            total.capacity += si.capacity;
+            total.free += si.free;
+          }
         }
-        iPos += (strlen( pcBuffer + iPos) + 1 );
-      }while( strlen( pcBuffer + iPos ) > 0 );
+        iPos += (strlen(pcBuffer + iPos) + 1);
+      } while (strlen(pcBuffer + iPos) > 0);
     }
     delete[] pcBuffer;
 #else // for linux and osx
-    if( GetDiskFreeSpaceEx( "/", NULL, &ULTotal, &ULTotalFree ) )
-    {
-      ULTotalTmp.QuadPart+= ULTotal.QuadPart;
-      ULTotalFreeTmp.QuadPart+= ULTotalFree.QuadPart;
-    }
+    total = space("/", ec);
```

Comment:
correct, there are a number of skins that use the System.FreeSpace / System.TotalSpace infolabels.

they are either shown in some home screen widget or in the filemanager window.